src/ts_catalog/continuous_agg.h

```diff
@@ -73,22 +73,33 @@ ts_continuous_agg_get_compression_defelems(const WithClauseResult *with_clauses)
  */
 typedef struct ContinuousAggsBucketFunction
 {
-	/*
-	 * Schema of the bucketing function.
-	 * Equals TRUE for "timescaledb_experimental", FALSE otherwise.
-	 */
-	bool experimental;
-	/* Name of the bucketing function, e.g. "time_bucket" or "time_bucket_ng" */
-	char *name;
-	/* `bucket_width` argument of the function */
+	/* Oid of the bucketing function. In the catalog table, the regprocedure is used. This ensures
+	 * that the Oid is mapped to a string when a backup is taken and the string is converted back to
+	 * the Oid when the backup is restored. This way, we can use an Oid in the catalog table even
+	 * when a backup is restored and the Oid may have changed. However, the dependency management in
+	 * PostgreSQL does not track the Oid. If the function is dropped and a new one is created, the
+	 * Oid changes and this value points to a non-existing Oid. This can not happen in real-world
+	 * situations since PostgreSQL protects the bucket_function from deletion until the CAgg is
+	 * defined. */
+	Oid bucket_function;
+
+	/* `bucket_width` argument of the function. */
 	Interval *bucket_width;
+
 	/*
 	 * Custom origin value stored as UTC timestamp.
 	 * If not specified, stores infinity.
 	 */
-	Timestamp origin;
-	/* `timezone` argument of the function provided by the user */
+	Timestamp bucket_origin;
```

Comment:
```suggestion
	TimestampTz bucket_origin;
```
Shouldn't this be `TimestampTz` instead?

Reply:
You are right; this variable should be converted into a `TimestampTz`. However, in this PR I only rename the variable. We have some existing logic that handles and converts the value, and we use the in and out functions for timestamps at the moment. I will open a follow-up PR with an isolated migration for this datatype.